### PR TITLE
fix: add jobs feature

### PR DESCRIPTION
**Pull request by lonniebe**

### CODE REVIEW

File: BE/lib/rest/job.py [hunk 1]
```diff
@@ -1,21 +1,50 @@
...
 
 router = APIRouter()
 
 
+@router.get(
+    "/jobs/{job_id}",
+    name="Get existing jobs",
+    description="Frontend Notice: You get all jobs and need to filter the relevance according to the User faculty ",
+    status_code=status.HTTP_200_OK,
+    response_model=Job
```
Comment (pazbechor) [inline BE/lib/rest/job.py:23]: Rename To JobOut

File: BE/lib/utils/enums.py [hunk 2]
```diff
@@ -17,7 +17,7 @@ class Experience(Enum):
 
 
 class Faculty(Enum):
-    ComputerScience = 'ComputerScience'
+    ComputerScience = "ComputerScience"
```
Comment (pazbechor) [inline BE/lib/utils/enums.py:20]: Why?

File: BE/lib/utils/db/models/job.py [hunk 3]
```diff
@@ -1,11 +1,19 @@
...
+from ...enums import JobTime, Faculty, Experience
 
 
 class Job(Base):
     __tablename__ = "job"
 
-
     id = Column(String, primary_key=True, unique=True, index=True)
+    publisher_email = Column(String, nullable=False)
+    published_time = Column(DateTime, nullable=False)
```
Comment (pazbechor) [inline BE/lib/utils/db/models/job.py:11]: make it default created like we doe everywhere else

File: BE/lib/rest/profile.py [hunk 4]
```diff
@@ -40,7 +40,6 @@ def get_profile(
     response_model=UserProfileOut
 )
 def get_profile(
-    user: Annotated[User, Depends(get_current_active_user)],
```
Comment (pazbechor) [inline BE/lib/rest/profile.py:43]: What? no. 
I already added  the endpoint:   "/profile/{user_email}" which return user profile. The Bearer Token stil need to be passed...else, everybody can get all profile exists in the system with no Token

**Comment by pazbechor** [comment]
Remove `Async` from all endpoints, add ` auth: Annotated[User, Depends(get_current_active_user)]` to all endpoints!

File: BE/lib/rest/job.py (hunk 1 above)
Comment (lonniebe) [reply BE/lib/rest/job.py:23]: done.

File: BE/lib/utils/enums.py (hunk 2 above)
Comment (lonniebe) [reply BE/lib/utils/enums.py:20]: reverted.

File: BE/lib/utils/db/models/job.py (hunk 3 above)
Comment (lonniebe) [reply BE/lib/utils/db/models/job.py:11]: done.

File: BE/lib/rest/profile.py (hunk 4 above)
Comment (lonniebe) [reply BE/lib/rest/profile.py:43]: reverted.

**Pull request merged**